src/pdb/mod.rs

```diff
@@ -325,50 +255,62 @@ pub struct Page {
+    #[br(parse_with = Self::parse_row_groups, args(num_rows, num_row_groups))]
+    pub row_groups: Vec<RowGroup>,
 }
 
 impl Page {
-    const HEADER_SIZE: usize = 0x28;
+    /// Size of the page header in bytes.
+    pub const HEADER_SIZE: usize = 0x28;
+
+    /// Parse the row groups at the end of the page.
+    fn parse_row_groups<R: Read + Seek>(
+        reader: &mut R,
+        ro: &ReadOptions,
+        args: (u16, u16),
+    ) -> BinResult<Vec<RowGroup>> {
+        let (num_rows, num_row_groups) = args;
+        if num_row_groups == 0 {
+            return Ok(vec![]);
+        }
 
-    /// Parses a page of a PDB file.
-    fn parse(page_data: &[u8]) -> IResult<&[u8], Page> {
-        // Signature (?)
-        let (input, _) = nom::bytes::complete::tag(b"\0\0\0\0")(page_data)?;
-        let (input, page_index) = PageIndex::parse(input)?;
-        let (input, page_type) = PageType::parse(input)?;
-        let (input, next_page) = PageIndex::parse(input)?;
-        let (input, unknown1) = nom::number::complete::le_u32(input)?;
-        let (input, unknown2) = nom::number::complete::le_u32(input)?;
-        let (input, num_rows_small) = nom::number::complete::u8(input)?;
-        let (input, unknown3) = nom::number::complete::u8(input)?;
-        let (input, unknown4) = nom::number::complete::u8(input)?;
-        let (input, page_flags) = nom::number::complete::u8(input)?;
-        let (input, free_size) = nom::number::complete::le_u16(input)?;
-        let (input, used_size) = nom::number::complete::le_u16(input)?;
-        let (input, unknown5) = nom::number::complete::le_u16(input)?;
-        let (input, num_rows_large) = nom::number::complete::le_u16(input)?;
-        let (input, unknown6) = nom::number::complete::le_u16(input)?;
-        let (input, unknown7) = nom::number::complete::le_u16(input)?;
+        let mut row_groups = Vec::with_capacity(num_row_groups.into());
 
-        let page = Page {
-            page_index,
-            page_type,
-            next_page,
-            unknown1,
-            unknown2,
-            num_rows_small,
-            unknown3,
-            unknown4,
-            page_flags,
-            free_size,
-            used_size,
-            unknown5,
-            num_rows_large,
-            unknown6,
-            unknown7,
-        };
+        // Calculate number of rows in last row group
+        let mut num_rows_in_last_row_group = num_rows % RowGroup::MAX_ROW_COUNT;
+        if num_rows_in_last_row_group == 0 {
+            num_rows_in_last_row_group = RowGroup::MAX_ROW_COUNT;
+        }
+
+        // Read last row group
+        let row_group = RowGroup::read_options(reader, ro, (num_rows_in_last_row_group,))?;
+        row_groups.push(row_group);
+
+        // Read remaining row groups
+        for _ in 1..num_row_groups {
+            let row_group = RowGroup::read_options(reader, ro, (RowGroup::MAX_ROW_COUNT,))?;
+            row_groups.insert(0, row_group);
+        }
```

Comment:
I dislike the fact that this is kinda brittle in regards to ordering and so forth. Also the repeated insertion at the front is not ideal but I don't see a good alternative either.

Reply:
I contemplated if I should use a `VecDeque` instead, but then decided against it. Didn't want to spend too much time on premature optimization, we can always fine-tune the code later.